query-engine/core/src/executor/interpreting_executor.rs

```diff
@@ -0,0 +1,66 @@
+use super::{pipeline::QueryPipeline, QueryExecutor};
+use crate::{
+    CoreResult, IrSerializer, QueryDocument, QueryGraph, QueryGraphBuilder, QueryInterpreter, QuerySchemaRef, Response,
+};
+use connector::{Connector, Result as ConnectorResult, TransactionLike};
+
+/// Central query executor and main entry point into the query core.
+/// Interprets the full query tree to return a result.
+pub struct InterpretingExecutor<C> {
+    connector: C,
+    primary_connector: &'static str,
+}
+
+// Todo:
+// - Partial execution semantics?
+// - ReadQueryResult + write query results should probably just be QueryResult
+impl<C> InterpretingExecutor<C>
+where
+    C: Connector + Send + Sync + 'static,
+{
+    pub fn new(connector: C, primary_connector: &'static str) -> Self {
+        InterpretingExecutor {
+            connector,
+            primary_connector,
+        }
+    }
+
+    pub fn with_interpreter<'a, F>(&self, f: F) -> ConnectorResult<Response>
```

Comment:
This might be nasty with async/await. Luckily we might not need this kind of arch soon... See https://github.com/prisma/prisma-engine/blob/bench/query-engine/connectors/sql-query-connector/src/transactional/mod.rs#L21

So instead of `with_transaction {}` we can just get a connection and start a transaction!